modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/GranuleDescriptor.java

```diff
@@ -137,14 +140,25 @@
 
     OverviewsController overviewsController;
 
+    private CoordinateReferenceSystem granuleCRS;
```

Comment:
Looks like this one is a dead store, not initialized, nor used anywhere. The CRS information is also most likely available in the envelope (barring bugs, the envelope should always have a CRS, unless the reader does not have one at all)